src/main/java/dk/kb/util/InjectionMavenPlugin.java

```diff
@@ -89,93 +86,17 @@ public void execute() throws MojoExecutionException {
      *          value is retrieved.
      */
     private String getYamlValueAsEnumFromFile(YamlResolver yamlInput) {
-        try {
-            switch (yamlInput.yamlType){
-                case "Sequence":
-                    return createEnumFromYamlValues(yamlInput);
-                case "List":
-                    return constructEnumFromLists(yamlInput);
-                case "Map":
-                    return constructEnumFromMap(yamlInput);
-                case "Single-value":
-                    return getSingleValue(yamlInput);
-                default:
-                    throw new MojoFailureException("Wrong key: '" + yamlInput.yamlType + "' has been used when configuring the plugin. " +
-                            "Valid keys are: 'Sequence', 'List', 'Map' and 'Single-value'.");
-            }
-
-        } catch (MojoFailureException e) {
-            throw new RuntimeException(e);
-        }
-    }
-
-    /**
-     * Create a string from all entries in a YAML map. The strings are concatenated by ', ' and
-     * can be used as an ENUM value in an OpenAPI specification.
-     * @param yamlInput a configuration object containing information needed to extract the values from the YAML.
-     * @return all values from the given map as a comma separated string.
-     */
-    private String constructEnumFromMap(YamlResolver yamlInput) {
         try {
             YAML fullYaml = YAML.resolveLayeredConfigs(filePath);
             StringJoiner stringJoiner = new StringJoiner(", ");
-            YAML propertiesMap = fullYaml.getSubMap(yamlInput.yamlPath);
-            propertiesMap.forEach((key, value) -> stringJoiner.add(value.toString()));
+            List<Object> properties = fullYaml.getMultiple(yamlInput.yamlPath);
+            properties.forEach(value -> stringJoiner.add((CharSequence) value));
             return stringJoiner.toString();
```

Comment:
Or maybe just
```Java
    return Strings.join(YAML.resolveLayeredConfigs(filePath).getMultiple(yamlInput.yamlPath));
```
or
```Java
    return Strings.join(cachedConfig.getMultiple(yamlInput.yamlPath));
```
if the suggestion on caching the YAML is followed? 

Yes, a lot of Java programming is knowing the Java API.